unit-tests/run-unit-tests.py

```diff
@@ -197,34 +229,39 @@ def cat( filename ):
     except FileNotFoundError:
         error( red + testname + reset + ': executable not found! (' + exe + ')' )
         continue
+    
+    check_log(log)
+    
 
-    # Normal logs are expected to have in last line:
-    #     "All tests passed (11 assertions in 1 test case)"
-    # Tests that have failures, however, will show:
-    #     "test cases: 1 | 1 failed
-    #      assertions: 9 | 6 passed | 3 failed"
-    for ctx in grep( r'^test cases:\s*(\d+) \|\s*(\d+) (passed|failed)', log ):
-        m = ctx['match']
-        total = int(m.group(1))
-        passed = int(m.group(2))
-        if m.group(3) == 'failed':
-            # "test cases: 1 | 1 failed"
-            passed = total - passed
-        if passed < total:
-            if total == 1  or  passed == 0:
-                desc = 'failed'
-            else:
-                desc = str(total - passed) + ' of ' + str(total) + ' failed'
+# adding given directory to python search path so that python tests can find pyrealsense2.pyd
+abs = os.path.abspath(dir)
+abs = abs.replace('\\' , '/')
+os.environ["PYTHONPATH"] = abs
 
-            if verbose:
-                error( red + testname + reset + ': ' + desc )
-                info( 'Executable:', exe )
-                info( 'Log: >>>' )
-                out()
-                cat( log )
-                out( '<<<' )
-            else:
-                error( red + testname + reset + ': ' + desc + '; see ' + log )
+# relative path to unit-test folder containing python tests
+unit_tests_dir = "../../unit-tests"
+
+for py_test in find(unit_tests_dir, '(^|/)test-.*\.py'):
+    
+    testdir = py_test[:-3]                         # "log/internal/test-all"  <-  "log/internal/test-all.py"
+    testparent = os.path.dirname(testdir)          # same as for cpp files
+    if testparent != "":
+        testname = 'test-' + testparent.replace( '/', '-' ) + '-' + os.path.basename(testdir)[5:]
+    else:
+        testname = 'test-' + os.path.basename(testdir)[5:]
+    
+    log = logdir + '/' + testname + '.log'
+
+    progress( testname, '>', log, '...' )
+    n_tests += 1
+    cmd = "python " + unit_tests_dir + '/' + py_test
```

Comment:
Need to make sure python 3 is used, if possible -- don't want python 2 to suddently mess things up.

Reply:
I changed the command to "py -3", that should run using python 3